src/main/java/org/marketdesignresearch/mechlib/mechanism/auctions/cca/bidcollection/supplementaryround/LastBidsTrueValueSupplementaryRound.java

```diff
@@ -13,23 +13,22 @@
 
 import java.util.Iterator;
 
-@ToString
-@EqualsAndHashCode
 public class LastBidsTrueValueSupplementaryRound implements SupplementaryRound {
 
     private static final int DEFAULT_NUMBER_OF_SUPPLEMENTARY_BIDS = 500;
 
     @Setter @Getter
     private int numberOfSupplementaryBids = DEFAULT_NUMBER_OF_SUPPLEMENTARY_BIDS;
-    private final Bids bids;
+
+    private final CCAuction auction;
```

Comment:
Better than removing ToString and EqualsAndHashCode would probably be to exclude this field via `[@ToString.Exclude](https://projectlombok.org/features/ToString)` and `[@EqualsAndHashCode.Exclude](https://projectlombok.org/features/EqualsAndHashCode)`. A good idea is also to make it at least java-`transient`, to avoid circular serialization.

Reply:
I do not agree that this would be a solution. Excluding this field from equal and hashcode would be wrong in this case. SupplementaryRounds for different auctions should not be equal in this implementation.
Moreover, making a field java-transient is only a good idea if the model can handle such a value beeing null. In this case a deserialized model with an auction == null would not work anymore and thus transient is here not a solution as well.

Reply:
Thanks a lot for raising the points, you're totally right of course!